accepted/single-file/design.md

```diff
@@ -66,225 +57,209 @@ Publishing to a single file can be triggered by adding the following property to
+
+### API Semantics
 
 #### `Assembly.Location`
 
-There are a few options to consider for the `Assembly.Location`  property of a bundled assembly:
+There are a few options to consider for the `Assembly.Location` property of a bundled assembly:
 
 * A fixed literal (ex: `null`) indicating that no actual location is available.
+* Throw an `AssemblyLoadedFromBundle` exception
+* The empty string, similar to assemblies loaded from [byte-array](https://docs.microsoft.com/en-us/dotnet/api/system.reflection.assembly.location?view=netcore-3.1#property-value),
 * The simple name of the assembly (with no path).
-* A special UNC notation such as `<bundle-path>/:/asm.dll` to denote files that come from the bundle. 
 * The path of the assembly as if it were not to be packaged into the single-file.
-* In the case of files spilled to disk (ex: ready-to-run assemblies) -- the actual location of the extracted file.
+* A special UNC notation such as `<bundle-path>/:/asm.dll` to denote files that come from the bundle. 
 * A configurable selection of the above, etc.
 
-We propose keeping the default behavior of `Assembly.Location`. That is, 
-
-* If the assembly is loaded directly from the bundle, return empty-string
-* If the assembly is extracted to disk, return the location of the extracted file.
+Proposed solution is for `Assembly.Location` to return the empty-string for bundled assemblies, which is the default behavior for assemblies loaded from memory.
 
 Most of the app development can be agnostic to whether the app is published as single-file or not. However, the parts of the app that deal with physical locations of files need to be aware of the single-file packaging. 
 
 #### `AppContext.BaseDirectory`
 
-A few options to consider for the `AppContext.BaseDirectory` are:
+`AppContext.BaseDirectory` will be the directory where the AppHost (the single-file bundle itself) resides. In contrast to [.NET Core 3.x single-file apps](design_3_0.md), .NET 5 single-file apps do not always self-extract on startup. Therefore, the details about extraction directory are not exposed through the `AppContext.BaseDirectory` API.  
 
-* The directory where embedded files are extracted out: This option enables easy access to content files spilled to disk. However, if no files need to be extracted to disk for an application bundle, there's no extraction directory. 
-* The extraction-directory if files are extracted, the AppHost directory otherwise. The limitation to this approach is that the value of  `AppContext.BaseDirectory`  changes with the way applications are packaged/executed.
-* The directory where `AppHost` binary resides (always).
+However, when single file apps are published with `IncludeAllContentInSingleFile` property set (which emulates .NET Core 3.x bundling behavior), `AppContext.BaseDirectory` returns the extraction directory, following  [.NET Core 3.x semantics](design_3_0.md#API-Impact).
```

Comment:
```suggestion
However, when single file apps are published with `IncludeAllContentInSingleFile` property set (which provides backward compatibility with .NET Core 3.x bundling behavior), `AppContext.BaseDirectory` returns the extraction directory, following  [.NET Core 3.x semantics](design_3_0.md#API-Impact).
```

Instead of "replicate" or "emulate", I think it is better say directly that this option is for .NET Core 3.x compat.